src/darwin/Framework/CHIPTests/MTRDeviceTests.m

```diff
@@ -1402,6 +1402,45 @@ - (void)test017_TestMTRDeviceBasics
         [subscriptionDroppedExpectation fulfill];
     };
 
+    // Before resubscribe, first test write failure and expected value effects
+    NSNumber * testEndpointID = @(1);
+    NSNumber * testClusterID = @(8);
+    NSNumber * testAttributeID = @(10000);
+    XCTestExpectation * expectedValueReportedExpectation = [self expectationWithDescription:@"Expected value reported"];
+    XCTestExpectation * expectedValueRemovedExpectation = [self expectationWithDescription:@"Expected value removed"];
+    delegate.onAttributeDataReceived = ^(NSArray<NSDictionary<NSString *, id> *> * attributeReport) {
+        for (NSDictionary<NSString *, id> * attributeDict in attributeReport) {
+            MTRAttributePath * attributePath = attributeDict[MTRAttributePathKey];
+            XCTAssertNotNil(attributePath);
+            if ([attributePath.endpoint isEqualToNumber:testEndpointID] && [attributePath.cluster isEqualToNumber:testClusterID] &&
+                [attributePath.attribute isEqualToNumber:testAttributeID]) {
+                id data = attributeDict[MTRDataKey];
+                if (data) {
+                    [expectedValueReportedExpectation fulfill];
+                } else {
+                    [expectedValueRemovedExpectation fulfill];
+                }
+            }
+        }
+    };
+
+    NSDictionary * writeValue = [NSDictionary
+        dictionaryWithObjectsAndKeys:@"UnsignedInteger", @"type", [NSNumber numberWithUnsignedInteger:200], @"value", nil];
+    [device writeAttributeWithEndpointID:testEndpointID
+                               clusterID:testClusterID
+                             attributeID:testAttributeID
+                                   value:writeValue
+                   expectedValueInterval:@(20)
```

Comment:
This is 20ms, right?

Reply:
Thank you for catching this. I've changed this to 20000 for 20s.